### DISCUSSION TITLE
add DCT

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/libvips#2058] [Ideas]
Things like blurhash need a simple DCT. See this issue for sample code https://github.com/libvips/ruby-vips/issues/235

**Comment by renchap** [comment]
Do you have any idea on what the API could look like?

**Comment by jcupitt** [comment]
Sorry, no, it would need some experimentation I imagine.

**Comment by jfcalvo** [comment]
This would be a great addition so we can implement blurhash.

**Comment by wmertens** [comment]
Is the DCT different from the fourier transform?

**Reply by jcupitt** [reply]
It's a simpler, faster FFT.

https://en.wikipedia.org/wiki/Discrete_cosine_transform
